jina/optimize/__init__.py

```diff
@@ -0,0 +1,231 @@
+import os
+from itertools import tee
+import json
+from pathlib import Path
+import shutil
+from ruamel.yaml import YAML
+
+from jina.flow import Flow
+from jina.helper import colored
+from jina.logging import default_logger as logger
+from type import IntegerParameter, load_optimization_parameters
+
+
+class FlowRunner:
+    def __init__(
+        self,
+        index_document_generator,
+        query_document_generator,
+        index_batch_size,
+        query_batch_size,
+        env_yaml=None,
+        workspace_env="JINA_WORKSPACE",
```

Comment:
why do not you get the workspace directly instead of the name of an environment variable?

Reply:
This allows us to use ENV variable which is the current general practise. We don't need it for optimizer as new yamls are generated for every trial with value of workspace instead of ENV variable. I was thinking of making flow runner a general class to be used for any flows and hence kept this ENV support.

Reply:
The fact that is general practice does not mean in my opinion it should be used inside the code. One thing is what the client may provide and the other is how we use it. If we need to facilitate things to the user I think it should be in a different layer not in this core object.